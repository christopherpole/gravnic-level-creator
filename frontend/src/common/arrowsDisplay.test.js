import React from 'react';
import { MOVE_UP, MOVE_DOWN, MOVE_LEFT, MOVE_RIGHT } from 'gravnic-game';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import ArrowsDisplay from './arrowsDisplay';

configure({ adapter: new Adapter() });

describe('<ArrowsDisplay />', () => {
  let props;

  beforeEach(() => {
    props = {
      arrows: [MOVE_UP, MOVE_DOWN, MOVE_LEFT, MOVE_RIGHT],
    };
  });

  it('Renders without exploding', () => {
    const arrowsDisplay = shallow(<ArrowsDisplay {...props} />);

    expect(arrowsDisplay).toHaveLength(1);
  });

  it('Matches the current snapshot', () => {
    const arrowsDisplay = shallow(<ArrowsDisplay {...props} />);

    expect(toJson(arrowsDisplay)).toMatchSnapshot();
  });
});
