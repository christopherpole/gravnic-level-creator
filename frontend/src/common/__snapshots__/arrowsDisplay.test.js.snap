// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArrowsDisplay /> Matches the current snapshot 1`] = `
<styled.ul
  className="arrows"
  justifyContent="left"
>
  <styled.li
    key="0"
    maxArrowWidth="auto"
  >
    <ArrowIcon
      direction="MOVE_UP"
      fillColor="white"
      size="100%"
    />
  </styled.li>
  <styled.li
    key="1"
    maxArrowWidth="auto"
  >
    <ArrowIcon
      direction="MOVE_DOWN"
      fillColor="white"
      size="100%"
    />
  </styled.li>
  <styled.li
    key="2"
    maxArrowWidth="auto"
  >
    <ArrowIcon
      direction="MOVE_LEFT"
      fillColor="white"
      size="100%"
    />
  </styled.li>
  <styled.li
    key="3"
    maxArrowWidth="auto"
  >
    <ArrowIcon
      direction="MOVE_RIGHT"
      fillColor="white"
      size="100%"
    />
  </styled.li>
</styled.ul>
`;
